File: thitracnghiemapp/lib/models/cau_hoi.dart
import 'chu_de.dart';

class CauHoi {
  final int id;
  final String noiDung;
  final String? hinhAnh;
  final String? amThanh;
  final String dapAnA;
  final String dapAnB;
  final String? dapAnC;
  final String? dapAnD;
  final String dapAnDung;
  final int chuDeId;
  final ChuDe? chuDe;

  const CauHoi({
    required this.id,
    required this.noiDung,
    this.hinhAnh,
    this.amThanh,
    required this.dapAnA,
    required this.dapAnB,
    this.dapAnC,
    this.dapAnD,
    required this.dapAnDung,
    required this.chuDeId,
    this.chuDe,
  });

  factory CauHoi.fromJson(Map<String, dynamic> json) {
    return CauHoi(
      id: json['id'] as int,
      noiDung: json['noiDung'] as String? ?? '',
      hinhAnh: json['hinhAnh'] as String?,
      amThanh: json['amThanh'] as String?,
      dapAnA: json['dapAnA'] as String? ?? '',
      dapAnB: json['dapAnB'] as String? ?? '',
      dapAnC: json['dapAnC'] as String?,
      dapAnD: json['dapAnD'] as String?,
      dapAnDung: json['dapAnDung'] as String? ?? '',
      chuDeId: json['chuDeId'] as int? ?? 0,
      chuDe: json['chuDe'] is Map<String, dynamic>
          ? ChuDe.fromJson(json['chuDe'] as Map<String, dynamic>)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'noiDung': noiDung,
      'hinhAnh': hinhAnh,
      'amThanh': amThanh,
      'dapAnA': dapAnA,
      'dapAnB': dapAnB,
      'dapAnC': dapAnC,
      'dapAnD': dapAnD,
      'dapAnDung': dapAnDung,
      'chuDeId': chuDeId,
      if (chuDe != null) 'chuDe': chuDe!.toJson(),
    };
  }
}

File: ThiTracNghiemApi/Models/CauHoi.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ThiTracNghiemApi
{
    public class CauHoi
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string NoiDung { get; set; } = string.Empty;
        public string? HinhAnh { get; set; }
        public string? AmThanh { get; set; }
        [Required]
        public string DapAnA { get; set; } = string.Empty;
        [Required]
        public string DapAnB { get; set; } = string.Empty;
        public string? DapAnC { get; set; }
        public string? DapAnD { get; set; }
        [Required]
        public string DapAnDung { get; set; } = string.Empty;  // e.g., "A"
        public int ChuDeId { get; set; }
        public ChuDe? ChuDe { get; set; }
        public ICollection<ChiTietKetQuaThi> ChiTietKetQuaThis { get; set; } = new List<ChiTietKetQuaThi>();
    }
}